### DISCUSSION TITLE
How to distingush out the Language Server Extensions?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#390] [Extension Development QnA] [answered]
We know the Language Server is provied by Extension form.
![image](https://user-images.githubusercontent.com/12727582/211140966-f333cbd9-49d0-40b0-9343-c641f67193e1.png)

But how to know which extensions belong to `Language Server` ?
![image](https://user-images.githubusercontent.com/12727582/211140905-b3b7c844-ce07-47e6-bf65-e9bba7965987.png)

Does this one belong to? if yes, where is the instruction ?

![image](https://user-images.githubusercontent.com/12727582/211141027-cde4a61c-2cca-4dc1-ad5b-8798b281addd.png)

**Comment by isidorn** [comment, marked as answer]
https://microsoft.github.io/language-server-protocol/implementors/servers/

**Reply by liaodalin19903** [reply]
You see the Language Server is not belong VSCode：
![architecture diagram](https://user-images.githubusercontent.com/12727582/211456768-2ca7a6fb-5da0-426d-9521-188b0393da84.png)

but however seems the Language Server runing as Extension(so it runs in Extension Host),for example `clangd` :
![image](https://user-images.githubusercontent.com/12727582/211456742-6d2068f4-127b-4968-9752-5ccb57176761.png)

so, there is contradictory, how to understand the architecture diagram?
